### PR TITLE
Update C++ sandbox-build instructions

### PR DESCRIPTION
**Pull request by johnkerl**
Context: #997

### CODE REVIEW

File: libtiledbsoma/README.md [hunk 1]
```diff
@@ -103,21 +103,10 @@ Examples:
 
 ## Notes
 
-A build system with `libfmt` and `libspdlog` installed may conflict with the required versions for `tiledbsoma`. If that is the case, this configuration is known-good on Ubuntu 22.04:
+A build system with `libfmt` and `libspdlog` installed may conflict with the required versions for `tiledbsoma`. Suggestion:
```
Comment (eddelbuettel) [inline libtiledbsoma/README.md:106]: It's unclear language and it is not entirely unclear what our recommendation should be. I for one have _always_ had system `libfmt*` and `libspdlog*` and I have not been bitten.  Should be maybe just remove the sentence here as well as the suggestion?  At TileDB Core, the recommendation always has been (paraphrasing) "if in doubt, use a superbuild".

**Review by eddelbuettel — COMMENTED** [review]
I added a comment.  It's a complicated topic, across all operating systems and versions and hardware variants :face_with_head_bandage:

File: libtiledbsoma/README.md (hunk 1 above)
Comment (johnkerl) [reply libtiledbsoma/README.md:106]: @eddelbuettel you've had those libraries separately installed and it's worked for you. So we should retain the _option_ of having them -- as well as the known-good logic.

@gspowley has had those libraries _not_ separately installed and it's worked for him. So we should retain that option.

@johnkerl (yours truly) tried it both your way and George's way and neither one worked for me -- in part due to the dangling spdlog CMake file left around after uninstall. That caused untold amounts of frustration and failure to resolve the issue, even with multiple smart people involved. Civic duty compels me to include the hard-learned lesson.

I'll update the wording to reflect both use-cases.

Comment (eddelbuettel) [reply libtiledbsoma/README.md:106]: I am trying to point at the bigger picture: you already delineated two to three distinct cases at just one Linux distro. Are we sure this generalizes? Are we sure it covers older distros? What about Conda, Brew, ...?  For as long as we cannot say anything _definitive_ I lean towards saying little.

Comment (johnkerl) [reply libtiledbsoma/README.md:106]: @eddelbuettel when I can be definitive, I will; as a second choice, I lean toward helpfulness and sharing known issues / hard-won lessons.

Comment (eddelbuettel) [reply libtiledbsoma/README.md:106]: I hear you and I appreciate the intent to be maximally helpful. But recall how what was once an off-hand comment by me on what worked here stuck around 'forever' (til you cleaned) in prominent position.  I think I learned over the years that saying less may be better than saying something that may end up being misleading to some, but it is far from clear cut.  And .... this was just a comment and not an objection.  Full steam ahead.

**Comment by johnkerl** [comment]
Updated wording @nguyenv @gspowley @eddelbuettel

**Review by gspowley — APPROVED** [review]

**Pull request merged**